src/vsr/sync.zig

```diff
@@ -29,28 +29,18 @@ pub const Stage = union(enum) {
 
         target: Target,
         trailers: Trailers = Trailers.initFill(.{}),
-        /// Our new superblock's `vsr_state.previous_checkpoint_id`.
-        /// Arrives with command=sync_free_set.
-        previous_checkpoint_id: ?u128 = null,
-        /// The checksum of the prepare corresponding to checkpoint_op.
-        /// Arrives with command=sync_client_sessions.
-        checkpoint_op_checksum: ?u128 = null,
 
         pub fn done(self: *const @This()) bool {
             for (std.enums.values(vsr.SuperBlockTrailer)) |trailer| {
                 if (!self.trailers.get(trailer).done) return false;
             }
-
-            assert(self.previous_checkpoint_id != null);
-            assert(self.checkpoint_op_checksum != null);
             return true;
         }
     };
 
     pub const UpdatingSuperBlock = struct {
         target: Target,
-        previous_checkpoint_id: u128,
-        checkpoint_op_checksum: u128,
+        checkpoint_state: vsr.CheckpointState,
```

Comment:
Looked at where we might be copying Stage as I think sizeof here is not tiny. Found 

```
                // We had a sync target already, but we discovered a newer one.
                // Re-sync to the new target.
                .requesting_trailers => |stage| {
```

I think we might want to use `|*stage|` there